admin/class-yoast-notification-center.php

```diff
@@ -54,74 +57,162 @@ public static function get() {
-			$json_notifications = json_decode( $transient_notifications, true );
+	/**
+	 * Register notifications of conditions
+	 *
+	 * This has to happen after the translations have been loaded.
+	 */
+	public function register_notifications() {
+		/** @var Yoast_Notification_Condition $condition */
+		foreach ( $this->notification_conditions as $condition ) {
+			$notification = $condition->get_notification();
+			// Make sure we are working with a proper Notification.
+			if ( false === ( $notification instanceof Yoast_Notification ) ) {
+				continue;
+			}
 
-			// Create Yoast_Notification objects.
-			if ( count( $json_notifications ) > 0 ) {
-				foreach ( $json_notifications as $json_notification ) {
-					$notifications[] = new Yoast_Notification( $json_notification['message'], $json_notification['options'] );
-				}
+			if ( $condition->is_met() ) {
+				$this->add_notification( $notification );
+			}
+			else {
+				// Remove dismissal so it will be shown next time.
+				$this->clear_dismissal( $notification );
 			}
 		}
-
-		return $notifications;
 	}
 
 	/**
-	 * Clear the cookie
+	 * Dismiss a notification
 	 */
-	private function remove_transient() {
-		delete_transient( self::TRANSIENT_KEY );
+	public static function ajax_dismiss_notification() {
+		$notification_center = self::get();
+
+		$notification_id = filter_input( INPUT_POST, 'notification' );
+		$notification    = $notification_center->get_notification_by_id( $notification_id );
+
+		if ( false === ( $notification instanceof Yoast_Notification ) ) {
+			die( '-1' );
+		}
+
+		if ( $notification_center->maybe_dismiss_notification( $notification ) ) {
+			die( '1' );
+		}
+
+		die( '-1' );
 	}
 
 	/**
-	 * Clear local stored notifications
+	 * Check if the user has dismissed a notification
+	 *
+	 * @param string   $dismissal_key The dismissal key.
+	 * @param null|int $user_id       User ID to check on.
+	 *
+	 * @return bool
 	 */
-	private function clear_notifications() {
-		$this->notifications = array();
+	public static function is_notification_dismissed( $dismissal_key, $user_id = null ) {
+		// Empty dismissal keys can never be set i.e. non-persistent notifications are always active.
+		if ( empty( $dismissal_key ) ) {
+			return false;
+		}
+
+		$user_id = ( ! is_null( $user_id ) ? $user_id : get_current_user_id() );
+
+		$current_value = get_user_meta( $user_id, $dismissal_key, $single = true );
+
+		return ! empty( $current_value );
 	}
 
 	/**
-	 * Remove transient when the plugin is deactivated
+	 * Check if the nofitication is being dismissed
+	 *
+	 * @param string|Yoast_Notification $notification Notification to check dismissal of.
+	 * @param string                    $meta_value   Value to set the meta value to if dismissed.
+	 *
+	 * @return bool True if dismissed.
 	 */
-	public function deactivate_hook() {
-		$this->clear_notifications();
+	public static function maybe_dismiss_notification( $notification, $meta_value = 'seen' ) {
```

Comment:
You can just call this `dismiss_notification`. It will dismiss it, unless impossible.